### PR TITLE
mostly clang-tidy cleanups

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

**Review by piponazo — CHANGES_REQUESTED** [review]
Thanks for the contribution! I just made some minor comments here and there, but most of the changes look good to me :wink:

File: src/basicio.cpp [hunk 1]
```diff
@@ -107,17 +107,17 @@ namespace Exiv2 {
...
-        HANDLE hFile_;                  //!< Duplicated fd
-        HANDLE hMap_;                   //!< Handle from CreateFileMapping
+        HANDLE hFile_{};  //!< Duplicated fd
+        HANDLE hMap_{};   //!< Handle from CreateFileMapping
 #endif
-        byte*  pMappedArea_;            //!< Pointer to the memory-mapped area
-        size_t mappedLength_;           //!< Size of the memory-mapped area
-        bool   isMalloced_;             //!< Is the mapped area allocated?
-        bool   isWriteable_;            //!< Can the mapped area be written to?
+        byte* pMappedArea_{};    //!< Pointer to the memory-mapped area
```
[the remark this answers is not in the document]
Comment (neheb) [reply src/basicio.cpp:117]: bool gets initialized to 0, which is false. Pointers gets initialized to nullptr. Note that smart pointers get default initialized to nullptr.

File: src/preview.cpp [hunk 2]
```diff
@@ -472,7 +472,7 @@ namespace {
             return {};
         }
         if (nativePreview_.filter_.empty()) {
-            return {data + nativePreview_.position_, static_cast<long>(nativePreview_.size_)};
+            return {data + nativePreview_.position_, static_cast<size_t>(static_cast<long>(nativePreview_.size_))};
```
[the remark this answers is not in the document]
Comment (neheb) [reply src/preview.cpp:475]: clang-tidy did this automatically. I'll fix it manually.

[the remark this answers is not in the document]
Comment (neheb) [reply src/preview.cpp:475]: Oh this may have been a bugfix. {} does not allow implicit conversions, hence the cast.

**Comment by neheb** [comment]
That should be it.

**Review by piponazo — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**